x/mongo/driver/topology/topology_options_test.go

```diff
@@ -9,73 +9,49 @@ package topology
 import (
 	"fmt"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/assert"
-	"go.mongodb.org/mongo-driver/x/mongo/driver/connstring"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-func TestOptionsSetting(t *testing.T) {
-	conf := &config{}
-	ssts := time.Minute
-	name := "test"
-	assert.Zero(t, conf.cs)
-
-	opt := WithConnString(func(connstring.ConnString) connstring.ConnString {
-		return connstring.ConnString{
-			ServerSelectionTimeout:    ssts,
-			ServerSelectionTimeoutSet: true,
-			AppName:                   name,
-		}
-
-	})
-
-	assert.NoError(t, opt(conf))
-
-	assert.Equal(t, ssts, conf.serverSelectionTimeout)
-
-	serverConf := newServerConfig(conf.serverOpts...)
-	assert.Equal(t, name, serverConf.appname, "expected appname to be: %v, got: %v", name, serverConf.appname)
-}
-
-func TestDirectConnectionFromConnString(t *testing.T) {
-	singleConnect := connstring.ConnString{
-		Connect:    connstring.SingleConnect,
-		ConnectSet: true,
-	}
-	autoConnect := connstring.ConnString{
-		Connect:    connstring.AutoConnect,
-		ConnectSet: true,
-	}
-	directConnectionTrue := connstring.ConnString{
-		DirectConnection:    true,
-		DirectConnectionSet: true,
-	}
-	directConnectionFalse := connstring.ConnString{
-		DirectConnection:    false,
-		DirectConnectionSet: true,
-	}
-	defaultConnString := connstring.ConnString{}
-
-	testCases := []struct {
-		name string
-		cs   connstring.ConnString
-		mode MonitorMode
-	}{
-		{"connect=direct", singleConnect, SingleMode},
-		{"connect=automatic", autoConnect, AutomaticMode},
-		{"directConnection=true", directConnectionTrue, SingleMode},
-		{"directConnection=false", directConnectionFalse, AutomaticMode},
-		{"default", defaultConnString, AutomaticMode},
-	}
-	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
-			topo, err := New(WithConnString(func(connstring.ConnString) connstring.ConnString { return tc.cs }))
-			assert.Nil(t, err, "topology.New error: %v", err)
-			assert.Equal(t, tc.mode, topo.cfg.mode, "expected mode %v, got %v", tc.mode, topo.cfg.mode)
-		})
-	}
-}
+//func TestDirectConnectionFromConnString(t *testing.T) {
```

Comment:
Is this test still needed? I'm not sure how exactly it would work with the new API.

Reply:
If it's redundant with a `connstring` or `options` package test (or unified spec test), then it seems safe to remove.